util/design/gen-otp-img.py

```diff
@@ -0,0 +1,183 @@
+                        default=MEMORY_HEXFILE,
+                        help='''
+                        Custom output path for generated hex file.
+                        Defaults to {}
+                        '''.format(MEMORY_HEXFILE))
+    parser.add_argument('--img-cfg',
+                        type=str,
+                        metavar='<path+filename>',
+                        default=IMAGE_DEFINITION_FILE,
+                        help='''
+                        Image configuration file in Hjson format.
+                        Defaults to {}
+                        '''.format(IMAGE_DEFINITION_FILE))
+    parser.add_argument('--add-cfg',
+                        type=str,
+                        metavar='<path+filename>',
+                        action='extend',
+                        nargs='+',
+                        help='''
+                        Additional image configuration file in Hjson format.
+
+                        This switch can be specified multiple times.
+                        Image configuration files are parsed in the same
+                        order as they are specified on the command line,
+                        and partition item values that are specified multiple
+                        times are overridden in that order.
+
+                        Note that seed values in additional configuration files
+                        are ignored.
+                        ''')
+
+    try:
+        args = parser.parse_args()
+
+        log.info('Loading LC state definition file {}'.format(
+            LC_STATE_DEFINITION_FILE))
+        with open(LC_STATE_DEFINITION_FILE, 'r') as infile:
+            lc_state_cfg = hjson.load(infile)
+        log.info('Loading OTP memory map definition file {}'.format(
+            MMAP_DEFINITION_FILE))
+        with open(MMAP_DEFINITION_FILE, 'r') as infile:
+            otp_mmap_cfg = hjson.load(infile)
+        log.info('Loading main image configuration file {}'.format(
+            args.img_cfg))
+        with open(args.img_cfg, 'r') as infile:
+            img_cfg = hjson.load(infile)
+
+        # If specified, override the seeds.
+        _override_seed(args, 'lc_seed', lc_state_cfg)
+        _override_seed(args, 'otp_seed', otp_mmap_cfg)
+        _override_seed(args, 'img_seed', img_cfg)
+
+        otp_mem_img = OtpMemImg(lc_state_cfg, otp_mmap_cfg, img_cfg)
+
+        print(args.add_cfg)
+
+        if args.add_cfg:
+            for f in args.add_cfg:
+                log.info(
+                    'Processing additional image configuration file {}'.format(
+                        f))
+                log.info('')
+                with open(f, 'r') as infile:
+                    cfg = hjson.load(infile)
+                    otp_mem_img.override_data(cfg)
+                log.info('')
+
+        # Print all defined args into header comment for reference
+        argstr = ''
+        arglist = vars(args)
```

Comment:
`vars` returns a dictionary. Maybe replace the next line with something like `for arg, argval in sorted(vars(args).items())` to get canonical sorting and avoid the `getattr`?

Reply:
good idea, done.